### ISSUE TITLE
Display languages for projects that have 2 or more repos

### ISSUE THREAD

**Issue by JessicaLucindaCheng** [hackforla/website#5160]
### Dependency
- [x] #5231 

### Overview
We need to change the code for displaying the language data so that it can display languages for projects that have 2 or more repositories (repos).

#### Developer
- [x] Understand how the website is currently getting languages data for projects [^1][^2]
- [x] In the _projects/civic-tech-index.md [^3], change `additional-repo-ids` from
  ```
  additional-repo-ids: '275296439'
  ```
  to
  ```
  additional-repo-ids:
    - 275296439
  ```
- [x] With the change you made above to `additional-repo-ids`, change the code to work with the 2 or more repos.
- [x] Check your code meets the Acceptance Criteria below

##### Acceptance Criteria
- [x] Code still works for any projects with only 1 repository (repo). 
    - [x] A project's markdown file that only has a non-empty `identification` field and no `additional-repo-ids` field.
    - [x] A project's markdown file with a non-empty `identification` field and an empty `additional-repo-ids` field.
- [x] Code for any projects with 2 or more repos 
  - [x] Possible scenario: A project has 2 repos, so the project's markdown file has a non-empty `identification` field and the `additional-repo-ids` field has 1 repo listed.
  - [x] Possible scenario: A project has 3 or more repos, so the project's markdown file has a non-empty `identification` field and an `additional-repo-ids` field with 2 or more repos listed.
- [x] When displaying languages for a project, the same languages are not listed multiple times. For example, if a project has HTML as a language in two repos, its list of languages when displayed on the website should only list HTML once.
- [x] On the Projects page [^4],
  - [x] The filters work properly
  - [x] The languages and contents on the cards remain the same as on the live website
- [x] After all the code changes, check the Languages listed for the detailed Civic Tech Index project page matches the one on the live website [^5] 
- [x] Also, check other project's detailed pages to make sure everything looks and works okay
- [x] Nothing is broken on the website due to your code changes

#### Merge Team
- [ ] Once the pull request associated with this issue is merged, go to #6138 and
  - [ ] Check off this issue under the _Dependency_ section
  - [ ] If all the issues in _Phase 1_ of the _Dependency_ section are checked off, move each of the issues in _Phase 2_ of the _Dependency_ section to the **New Issue Approval** column and apply a **ready for prioritization** label to each of those issues.

### Resources/Instructions
- This issue is tracked in #6138 
[^1]: [Wiki: How languages are added to the `project` variable](https://github.com/hackforla/website/wiki/Guide-to-the-Project-Layout-File#adding-languages)
[^2]: Previous issue #3805 and its pr #5037 that the work for this issue should be built upon. (Note: Due to code changes, please look at the actual files in the codebase for the most current code.) Also, look at #5231 for the refactor of variables and fields. 
[^3]: [_projects/civic-tech-index.md](https://github.com/hackforla/website/blob/gh-pages/_projects/civic-tech-index.md?plain=1)
[^4]: [Projects page](https://www.hackforla.org/projects/)
[^5]: [Civic Tech Index page](https://www.hackforla.org/projects/civic-tech-index)

**Comment by StephenTheDev1001** [comment]
I'm guessing it should look this way if we're using more than one additional repo ID:

```
additional-repo-ids: 
  - '275296439'
  - '221542759'
```
When I try to access it in the [project.js](https://github.com/hackforla/website/blob/gh-pages/assets/js/project.js) file I'm getting it as one string ``` 275296439221542759 ```

From my initial research, we can't separate them by length because github repo IDs have different lengths. I originally wanted to do it so more than one additional repo id would work, but this was one of my blockers.

What can be done to the [project.js](https://github.com/hackforla/website/blob/gh-pages/assets/js/project.js) and [current-projects.js](https://github.com/hackforla/website/blob/gh-pages/assets/js/current-projects.js) to make the additional repo ids accessible there?

**Comment by JessicaLucindaCheng** [comment]
> I'm guessing it should look this way if we're using more than one additional repo ID:
> 
> ```
> additional-repo-ids: 
>   - '275296439'
>   - '221542759'
> ```
> 
> When I try to access it in the [project.js](https://github.com/hackforla/website/blob/gh-pages/assets/js/project.js) file I'm getting it as one string `275296439221542759`
> 
> From my initial research, we can't separate them by length because github repo IDs have different lengths. I originally wanted to do it so more than one additional repo id would work, but this was one of my blockers.
> 
> What can be done to the [project.js](https://github.com/hackforla/website/blob/gh-pages/assets/js/project.js) and [current-projects.js](https://github.com/hackforla/website/blob/gh-pages/assets/js/current-projects.js) to make the additional repo ids accessible there?

I'm not exactly sure what your specific blocker was but I would try the following:
- Remove quotes from the id numbers, so
```
additional-repo-ids: 
  - 275296439
  - 221542759
```
- In assets/js/current-projects.js, use jsonify when creating the additional-repo-ids portion for the project object.
- In _layouts/project.html, use jsonify when creating the script attribute for additional-repo-ids. Then, in assets/js/project.js, use JSON.parse when getting the attribute.

That's the gist of it. I left out a lot of details, such as checks and additional code, that would need to be implemented to make it work. Also, there could be problems with what I suggested that I might be missing since I haven't implemented it and thoroughly tested it.

[^1]:  Search for "jsonify" in https://jekyllrb.com/docs/liquid/filters/

**Comment by JessicaLucindaCheng** [comment]
@StephenTheDev1001 I saw that your comment was hidden and wasn't sure if you still wanted a response but my response is at the bottom of this comment.

> I'm guessing it should look this way if we're using more than one additional repo ID:
> 
> ```
> additional-repo-ids: 
>   - '275296439'
>   - '221542759'
> ```
> 
> When I try to access it in the [project.js](https://github.com/hackforla/website/blob/gh-pages/assets/js/project.js) file I'm getting it as one string `275296439221542759`
> 
> From my initial research, we can't separate them by length because github repo IDs have different lengths. I originally wanted to do it so more than one additional repo id would work, but this was one of my blockers.
> 
> What can be done to the [project.js](https://github.com/hackforla/website/blob/gh-pages/assets/js/project.js) and [current-projects.js](https://github.com/hackforla/website/blob/gh-pages/assets/js/current-projects.js) to make the additional repo ids accessible there?

I'm not exactly sure what your specific blocker was but I would try the following:
- Remove quotes from the id numbers, so
```
additional-repo-ids: 
  - 275296439
  - 221542759
```
- In assets/js/current-projects.js, use jsonify when creating the additional-repo-ids portion for the project object.
- In _layouts/project.html, use jsonify when creating the script attribute for additional-repo-ids. Then, in assets/js/project.js, use JSON.parse when getting the attribute.

That's the gist of it. I left out a lot of details, such as checks and additional code, that would need to be implemented to make it work. Also, there could be problems with what I suggested that I might be missing since I haven't implemented it and thoroughly tested it.

[^1]:  Search for "jsonify" in https://jekyllrb.com/docs/liquid/filters/

**Comment by JessicaLucindaCheng** [comment]
@hackforla/website-pm Is there anything that you think needs to be added under the "Acceptance Criteria"? If there is nothing that needs to be added, please prioritize the issue. Thank you.

**Comment by jphamtv** [comment]
Availability: Monday-Friday 8–11 PM PST
Estimated ETA: 2/6/24

**Comment by jphamtv** [comment]
Progress Update:
1. Progress: Testing to ensure updated code meets the Acceptance Criteria
2. Blockers: None
3. Availability: Mon—Fri 8-11pm (PST)
4. ETA: 2/5/24

**Comment by jphamtv** [comment]
Progress Update:

Progress: Complete. Pull request awaiting review #6255 
Blockers: None
Availability: Mon—Fri 8-11pm (PST)
